crates/biome_js_analyze/src/syntax/correctness/no_initializer_with_definite.rs

```diff
@@ -0,0 +1,51 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic};
+
+use biome_diagnostics::category;
+use biome_js_syntax::{JsSyntaxKind, TextRange, TsDefiniteVariableAnnotation};
+use biome_rowan::AstNode;
+
+declare_rule! {
+    /// Disallow initializing a variable with a definite assertion to prevent `SyntaxError`.
+    ///
+    /// ## Examples
+    ///
+    /// ```js
+    /// let foo!: string = "bar";
+    /// ```
+    pub(crate) NoInitializerWithDefinite {
+        version: "1.0.0",
+        name: "noInitializerWithDefinite",
+    }
+}
+
+impl Rule for NoInitializerWithDefinite {
+    type Query = Ast<TsDefiniteVariableAnnotation>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        if let Some(parent) = node.clone().into_syntax().parent().clone() {
+            if parent.kind() == JsSyntaxKind::JS_VARIABLE_DECLARATOR {
+                if let Some(initializer) = parent
+                    .children()
+                    .find(|sibling| sibling.kind() == JsSyntaxKind::JS_INITIALIZER_CLAUSE)
+                {
+                    return Some(initializer.text_range());
+                }
+            }
+        }
+        None
```

Comment:
I think you can simplify at the following code and avoid one clone:
```suggestion
        node.parent::<JsVariableDeclarator>()
            .and_then(|var_declarator| var_declarator.initializer())
            .map(|init| init.text_range())
```

Reply:
Quick question, what is this type of syntax with the `::<Type>` called in Rust?

Reply:
I guess it's called explicit type annotation.

Reply:
> I guess it's called turbofish.
> 
> [rust-lang/rust@`7fd15f0`/src%2Ftest%2Fui%2Fparser%2Fbastion-of-the-turbofish.rs](https://github.com/rust-lang/rust/blob/7fd15f09008dd72f40d76a5bebb60e3991095a5f/src%2Ftest%2Fui%2Fparser%2Fbastion-of-the-turbofish.rs)

Ok that makes sense, thank you!